### ISSUE TITLE
Tcp-based connection?

### ISSUE THREAD

**Issue by knocte** [edjCase/JsonRpc#37]
Judging by briefly looking at the code (e.g. https://github.com/edjCase/JsonRpc/blob/master/src/EdjCase.JsonRpc.Client/RpcClient.cs), seems like EdjCase.JsonRpc libraries are entirely HTTP-based, instead of transport-agnostic JsonRpc-based. Am I right?

For example, I cannot manage to do the equivalent of this nc command below:

```
$ nc gh05.geekhosters.com 50001
{ "id": 0, "method": "server.version", "params": [ "1.9.5", "0.6" ] }
```

The result of that in a Linux terminal, is:

```
{"id": 0, "result": "1.0"}
```

Trying the same with EdjCase I get:

```
Unhandled Exception: System.AggregateException: One or more errors occurred. ---
> EdjCase.JsonRpc.Client.RpcClientUnknownException: Error occurred when trying t
o send rpc requests(s) ---> System.Net.Http.HttpRequestException: An error occur
red while sending the request. ---> System.Net.WebException: The server committe
d a protocol violation. Section=ResponseStatusLine
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
...
```

Presumably, because I had anyway to use a fake "http://" scheme before the hostname to construct a Uri object first.

Any chance EdjCase can be easily adapted for this use case as well?

**Comment by Gekctek** [comment]
Correct. It is dependent on http. I'll look into abstracting out the http context and making it usable through any protocol

**Comment by risogolo** [comment]
If you make it also work as a subprotocol and make it work over websockets, that would be also cool.

**Comment by knocte** [comment]
Couldn't wait for this so in the end I wrote an implementation myself using TcpClient in F#, will share it soon.

**Comment by Gekctek** [comment]
Very cool. Been very busy as of late, but I still plan on fixing this issue
